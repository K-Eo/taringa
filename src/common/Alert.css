.alert {
  border: none !important;
  border-radius: 0 !important;
}

.Alert-icon {
  margin-right: 16px;
}
