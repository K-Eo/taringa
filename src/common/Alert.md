Basic Alert

```js
<Alert>
  I am an alert
</Alert>
```

Basic Alert with icon

```js
<Alert icon="check">
  I am an alert
</Alert>
```

Success Alert

```js
<Alert type="success">
  I am an alert
</Alert>
```

Danger Alert

```js
<Alert type="danger">
  I am an alert
</Alert>
```

Warning Alert

```js
<Alert type="warning">
  I am an alert
</Alert>
```

Info Alert

```js
<Alert type="info">
  I am an alert
</Alert>
```
