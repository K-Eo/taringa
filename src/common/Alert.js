import React from 'react'
import classNames from 'classnames'

import './Alert.css'

const Alert = ({ type, className, ...props }) => {
  const wrapperClass = classNames('alert', className, {
    [`alert-${type}`]: true,
  })

  return (
    <div className={wrapperClass} {...props}>
      {props.icon && <i className="fa fa-check Alert-icon" />}
      {props.children}
    </div>
  )
}

export default Alert
