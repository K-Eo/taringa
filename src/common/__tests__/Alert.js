import React from 'react'
import renderer from 'react-test-renderer'

import Alert from '../Alert'

describe('Alert', () => {
  it('renders default', () => {
    const tree = renderer.create(<Alert>I am an alert</Alert>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with icon', () => {
    const tree = renderer
      .create(<Alert icon="check">I am an alert</Alert>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with type', () => {
    const tree = renderer
      .create(
        <Alert type="success" icon="check">
          I am an alert
        </Alert>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
