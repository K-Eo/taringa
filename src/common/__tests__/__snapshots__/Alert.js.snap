// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders default 1`] = `
<div
  className="alert alert-undefined"
>
  I am an alert
</div>
`;

exports[`Alert renders with icon 1`] = `
<div
  className="alert alert-undefined"
  icon="check"
>
  <i
    className="fa fa-check Alert-icon"
  />
  I am an alert
</div>
`;

exports[`Alert renders with type 1`] = `
<div
  className="alert alert-success"
  icon="check"
  type="success"
>
  <i
    className="fa fa-check Alert-icon"
  />
  I am an alert
</div>
`;
